src/tpm2_wrap.c

```diff
@@ -2560,9 +2602,19 @@ int wolfTPM2_NVCreateAuth(WOLFTPM2_DEV* dev, WOLFTPM2_HANDLE* parent,
         return rc;
     }
 
+    /* Compute NV Index name in case of parameter encryption */
+    rc = TPM2_HashNvPublic(&in.publicInfo.nvPublic,
+                            (byte*)&nv->handle.name.name,
+                            &nv->handle.name.size);
+    if (rc != TPM_RC_SUCCESS) {
+        return rc;
+    }
+
     /* return new NV handle */
+    XMEMSET(nv, 0, sizeof(*nv));
```

Comment:
This is broken. It will memset and clear the `nv->handle.name` done above. The memset needs moved above `TPM2_HashNvPublic`.

Reply:
Indeed, leftover after making the Hash Name computation a generic function. 

This actually had no impact over the examples, because we are making the Name computation here just in case some later command that will be using parameter encryption needs it.

Good catch!